components/content_settings/renderer/brave_content_settings_agent_impl.cc

```diff
@@ -311,6 +311,14 @@ bool BraveContentSettingsAgentImpl::IsFirstPartyCosmeticFilteringEnabled(
   return setting == CONTENT_SETTING_BLOCK;
 }
 
+void BraveContentSettingsAgentImpl::DidCommitProvisionalLoad(
+    ui::PageTransition transition) {
+  ContentSettingsAgentImpl::DidCommitProvisionalLoad(transition);
+  // Invalidate Ephemeral Storage opaque origins. Page reload might change the
+  // Ephemeral Storage mode, in this case we should re-request it.
+  cached_ephemeral_storage_origins_.clear();
```

Comment:
this is similar to upstream cleanup on the same event
https://source.chromium.org/chromium/chromium/src/+/main:components/content_settings/renderer/content_settings_agent_impl.cc;l=421-425;drc=a6bdc8f2993883fc55eb9cb0945694299b056675?q=ContentSettingsAgentImpl::DidCommitProvisionalLoad&ss=chromium%2Fchromium%2Fsrc